rotkehlchen/globaldb/updates/parsers.py

```diff
@@ -0,0 +1,334 @@
+import abc
+import logging
+import re
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+from rotkehlchen.assets.types import AssetData, AssetType
+from rotkehlchen.errors.asset import UnknownAsset
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address
+from rotkehlchen.types import ChainID, ChecksumEvmAddress, EvmTokenKind, Timestamp
+
+from .types import ParsedAssetData, VersionRange
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from rotkehlchen.db.drivers.gevent import DBConnection
+
+T = TypeVar('T')
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BaseParser(abc.ABC, Generic[T]):
+    """Base parser with common parsing functionality.
+
+    This was introduced to handle asset updates that need to be applied
+    before global db schema version changes. By having dedicated parser classes for different
+    data types (assets, collections, mappings), we can ensure compatible updates
+    are applied before the schema transitions to a new version.
+    """
+    def __init__(self) -> None:
+        self._double_quotes_re = re.compile(r'\"(.*?)\"')
+        self._string_re = re.compile(r'.*([\'"])(.*?)\1.*')
+        self._version_parsers: list[tuple[VersionRange, Callable]] = []
+
+    def parse(self, insert_text: str, version: int, connection: 'DBConnection') -> T:
+        for version_range, parser in self._version_parsers:
+            if version_range.contains(version):
+                return parser(insert_text=insert_text, connection=connection)
+
+        raise ValueError(f'No parser found for version {version}')
```

Comment:
We never ever use ValueErrors.

Use something else. Who handles it? Who should catch it? What should happen if this happens? When can this happen?